cmd/convertor/builder/builder_utils.go

```diff
@@ -0,0 +1,227 @@
+/*
+   Copyright The Accelerated Container Image Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package builder
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"io"
+	"os"
+	"path"
+
+	"github.com/containerd/accelerated-container-image/pkg/snapshot"
+	"github.com/containerd/containerd/archive/compression"
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/remotes"
+	"github.com/containerd/continuity"
+	"github.com/opencontainers/go-digest"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func fetchManifestAndConfig(ctx context.Context, fetcher remotes.Fetcher, manifestDesc specs.Descriptor) (specs.Manifest, specs.Image, error) {
+	// get manifest
+	rc, err := fetcher.Fetch(ctx, manifestDesc)
+	if err != nil {
+		return specs.Manifest{}, specs.Image{}, errors.Wrapf(err, "failed to fetch manifest")
+	}
+	buf, err := io.ReadAll(rc)
+	if err != nil {
+		return specs.Manifest{}, specs.Image{}, errors.Wrapf(err, "failed to fetch manifest")
```

Comment:
Nit: "failed to read manifest" to differentiate from the above?

Reply:
done